presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -94,6 +94,7 @@
 import io.prestosql.sql.planner.iterative.rule.RemoveFullSample;
```

Comment:
Also, DistinctLimit with limit higher than cardinality of its source node can be rewritten to DistinctNode

Reply:
Will merge that feature to `RemoveSingleRowDistinctLimit`